code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -329,15 +332,13 @@
 /obj/item/weapon/reagent_containers/food/snacks/donut/New()
 	..()
 	if(randomized_sprinkles && prob(30))
-		icon_state = "donut2"
-		name = "frosted donut"
-		reagents.add_reagent("sprinkles", 2)
-		filling_color = "#FF69B4"
+		new /obj/item/weapon/reagent_containers/food/snacks/donut/sprinkles(loc)
+		qdel(src)
```

Comment:
Not ideal, but it was a fix for an issue I encountered when testing the machine with the box of donuts the kitchen starts with. Frosted donuts in the box are simply renamed donuts, not actually a frosted donut pathed item. This was more of a band-aid fix for that than meant to be a lasting fix, while I came up with a better way of handling it. I'll put some more thought on to this to hopefully fix that better.